### ISSUE TITLE
Redirects to https domains break in 1.40

### ISSUE THREAD

**Issue by arsduo** [lynndylanhurley/devise_token_auth#832]
Thanks for the great gem! I discovered an error this week: #765 breaks HTTPS redirect URLs. 

In `app/controllers/devise_token_auth/omniauth_callbacks_controller.rb` (line 17), the new URL is constructed using `URI::HTTP.build`. According to the [documentation](http://ruby-doc.org/stdlib-2.3.1/libdoc/uri/rdoc/URI/HTTP.html#method-c-build) for `URI::HTTP`, though, `build` does not accept the `scheme` parameter that the code passes in. 

Reproduction:

```
# "https" is the value returned by Rack::Request for scheme when appropriate (see http://www.rubydoc.info/gems/rack/Rack/Request/Helpers#scheme-instance_method) 
URI::HTTP.build(scheme: request.scheme, host: request.host, port: request.port, path: path).to_s

URI::HTTP.build(scheme: "https", host: "google.com", port: 443, path: "/").to_s
=> "http://google.com:443/"
```

I'll make a PR to fix this later this week if I have a chance.

**Comment by arcreative** [comment]
Ahmagawd, THANK YOU.  I was tearing my hair out all day thinking it was my fault.  Reverting to `0.1.39` fixed it for now... Going to go drink heavily now 🍻

**Comment by zachfeldman** [comment]
@arsduo we'd love your PR! Feel free to submit.

**Comment by zachfeldman** [comment]
...otherwise we'll close this out in 7 days

**Comment by arcreative** [comment]
...but he didn’t break it ;-)  This is a pretty big issue, and a hard one to track down, at that...

**Comment by zachfeldman** [comment]
@arcreative this is a volunteer-run project, so if something needs to get done, it's on people that use this gem to implement it. That's why I asked if he could write a PR. You can also feel free to submit one if you'd like! We're going to try to much more actively monitor open pull requests and prioritize issues for the community to take on. I'm also going to try to write some code in addition to maintaining. But nobody is paid or is paying anyone to maintain this gem.

**Comment by zachfeldman** [comment]
....actually going to leave this open because there was at least a comment. @arcreative we await your PR! :)

**Comment by arcreative** [comment]
This seems to be fixed via 5300ea7.  Feel free to close, thanks!

**Comment by zachfeldman** [comment]
Great!
